pylint/test/test_self.py

```diff
@@ -150,6 +150,9 @@ def test_no_ext_file(self):
     def test_w0704_ignored(self):
         self._runtest([join(HERE, 'input', 'ignore_except_pass_by_default.py')], code=0)
 
+    def test_list_msgs(self):
+        self._runtest(['--list-msgs'], code=0)
+
```

Comment:
This test needs the reporter test deletions as per my `fergal-comments` branch in order to make tests run enough to select it. Didn't included the changes in both branches as tests aren't passing for per_dir_config anyway.